### DISCUSSION TITLE
Is there any fastlane action which support only exportArchive

### DISCUSSION THREAD

**Discussion by ykhandelwal913** [fastlane/fastlane#21869] [General]
I am trying to run the exportArchive using xcodebuild command and we need to change to rvm use sytem to make that work. Sometime it throws error like `rvm use system
23:42:57  RVM is not a function, selecting rubies with 'rvm use ...' will not work.
23:42:57  
23:42:57  You need to change your terminal emulator preferences to allow login shell.
23:42:57  Sometimes it is required to use `/bin/bash --login` as the command.
23:42:57  Please visit https://rvm.io/integration/gnome-terminal/ for an example.`

and if i dont use system ruby it fails with `error: exportArchive: ipatool failed
23:46:50  
23:46:50  AppThinning.StubError(errorDescription: Optional("ipatool failed"))`

Is there any fastlane action available to run only exportArchive with the params or any other way to do?

**Comment by florence0239** [comment]
Hello,

To run the exportArchive command using Fastlane, you can use the gym action, which is designed to build and package iOS apps. The gym action handles the xcodebuild commands internally, including exportArchive. Here’s how you can use it:

Install Fastlane if you haven’t already:
` sudo gem install fastlane -NV

`
Create a Fastfile in your project directory if you don’t have one:
`fastlane init

`
Configure the gym action in your Fastfile:
`lane :build do
  gym(
    workspace: "YourApp.xcworkspace",
    scheme: "YourAppScheme",
    export_method: "app-store", # or "ad-hoc", "enterprise", "development"
    export_options: {
      provisioningProfiles: {
        "com.your.bundle.id" => "Your Provisioning Profile Name"
      }
    }
  )
end`

Run the lane:
`fastlane build

`
This setup should handle the exportArchive process for you. If you encounter issues with RVM and Ruby versions, you can specify the Ruby version in your Fastfile or use a shell script to ensure the correct environment is set up before running Fastlane.


Hope this work for you.
Best Regard,
florence0239
[Official Website](https://www-choiceadvantage.com)
